go-example/metrics/gooteltest/cmd/oteltester/main.go

```diff
@@ -208,12 +204,18 @@ func main() {
 	if err != nil {
 		log.Fatalf("Error opening config file: %v", err)
 	}
+
+	prefix := "cum_"
+	if config.AggregationTemporalitySelector == "delta" {
+		prefix = "delta_"
+	}
+
 	shutdown := initMetric(config)
 	defer shutdown()
 	engine := gooteltest.NewEngine(config.ValueSets)
 	meter := global.Meter("opamp")
-	registerMetricObservers(meter, config.Metrics, engine)
-	registerHistograms(config.CollectPeriod, meter, config.Metrics, engine)
+	registerMetricObservers(meter, config.Metrics, engine, prefix)
+	registerHistograms(config.CollectPeriod, meter, config.Metrics, engine, prefix)
```

Comment:
Consider passing the config itself like this to cut down on the number of parameters:

registerHistograms(config, meter, engine)

Do the same for registerMetricObservers:

registerMetricObservers(meter, config, engine)